### PR TITLE
Add Frontend endpoint

### PR DESCRIPTION
**Pull request by wakwak3125**
## WHAT
refs: https://github.com/creasty/torinos/issues/12#issuecomment-312419999

- Create scaffold of all routeing.
  - [x] Create all routes.
  - [x] Create all empty  handler.

@yamadayuki @creasty

### CODE REVIEW

**Comment by wakwak3125** [comment]
@creasty I'd like you to review.

File: server/route/notification.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+// CreateNotification craete noitication of project
+func CreateNotification(c *gin.Context) {
+	uuid := c.Param("uuid")
+	c.String(http.StatusOK, "Create notification:"+uuid)
```
Comment (yamadayuki) [inline server/route/notification.go]: I expect the response to be as JSON. Will all endpoints response be to be with `Content-Type: application/json`?

Comment (wakwak3125) [reply server/route/notification.go]: Okay💪

File: server/route.go [hunk 2]
```diff
@@ -24,6 +24,13 @@ func drawRoutes(r *gin.Engine, appContext *system.AppContext) {
 	{
 		r.GET("/", route.Ping)
 		r.GET("/ping", route.Ping)
+		r.GET("/user", route.CurrentUser)
```
Comment (creasty) [inline server/route.go]: Use verb for function name.
e.g., `GetCurrentUser`, `Authorize`, etc

Comment (wakwak3125) [reply server/route.go]: Okay💪

**Comment by wakwak3125** [comment]
@creasty @yamadayuki 
I applied your review 👍

File: server/route.go [hunk 3]
```diff
@@ -24,6 +24,13 @@ func drawRoutes(r *gin.Engine, appContext *system.AppContext) {
 	{
 		r.GET("/", route.Ping)
 		r.GET("/ping", route.Ping)
+		r.GET("/user", route.GetCurrentUser)
+		r.GET("/oauth/github/authorization", route.Authorize)
+		r.POST("/oauth/github/authentication", route.Authenticate)
+		r.POST("/projects", route.Analyze)
```
Comment (creasty) [inline server/route.go]: How about `CreateProject` and `GetProject` for clarity

File: server/route/oauth.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+// Authorize user
+func Authorize(c *gin.Context) {
+	c.JSON(http.StatusOK, gin.H{
+		"message": "Authorize",
+	})
+}
+
+// Authenticate user
+func Authenticate(c *gin.Context) {
```
Comment (yamadayuki) [inline server/route/oauth.go:17]:
```go
// Authorize user
func Authorize(c *gin.Context) { ... }

// Authenticate user
func Authenticate(c *gin.Context) { ... }
```

I supposed these route functions above are little ambiguous. I think `Authorize()` should be `GetAuthorization() / GetAuthentication()`. How about this?

File: server/route/notification.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package route
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+// CreateNotification craete noitication of project
```
Comment (creasty) [inline server/route/notification.go]: Nitpicking.. there's typo

- cr**ae**te → cr**ea**te
- no**it**ication → no**tif**ication

And seemingly you always forget "三単現のs"...

> CreateNotification create**s** notification of project

Speaking of which, this sounds like actually sending notification instantly..
It's my bad to call this model "Notification" in the first place; On second thought, it should be "Subscription."

Comment (wakwak3125) [reply server/route/notification.go]: > And seemingly you always forget "三単現のs"...

(｡>﹏<｡)

File: server/route/oauth.go (hunk 4 above)
Comment (wakwak3125) [reply server/route/oauth.go:17]: OK!

File: server/route.go (hunk 3 above)
Comment (wakwak3125) [reply server/route.go]: OK!

**Comment by wakwak3125** [comment]
@creasty I fixed fucking typos and rename handler functions.

File: server/route.go [hunk 6]
```diff
@@ -25,12 +25,12 @@ func drawRoutes(r *gin.Engine, appContext *system.AppContext) {
 		r.GET("/", route.Ping)
 		r.GET("/ping", route.Ping)
 		r.GET("/user", route.GetCurrentUser)
-		r.GET("/oauth/github/authorization", route.Authorize)
-		r.POST("/oauth/github/authentication", route.Authenticate)
+		r.GET("/oauth/github/authorization", route.GetAuthorization)
+		r.POST("/oauth/github/authentication", route.GetAuthentication)
```
Comment (yamadayuki) [inline server/route.go]: Nope, `r.POST("...", route.Authenticate)` remains unchanged. This action is handled by `POST` request. I only suggest the `route.Authorize` handler should be appropriate name as a verb indicating `GET` request handler.

File: server/route/projects.go [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package route
```
Comment (creasty) [inline server/route/projects.go]: server/route/projects.go → server/route/project.go

File: server/route/notification.go [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package route
```
Comment (creasty) [inline server/route/notification.go]: server/route/notification.go → server/route/subscription.go

**Review by creasty — APPROVED** [review]
Lookin good! 🐻

**Pull request merged**